pkg/scheduler/api/helpers.go

```diff
@@ -33,15 +34,22 @@ func PodKey(pod *v1.Pod) TaskID {
 }
 
 func getTaskStatus(pod *v1.Pod) TaskStatus {
+	var gracePeriodSeconds int64 = 30
+	if pod.Spec.TerminationGracePeriodSeconds != nil {
+		// default grace period
+		gracePeriodSeconds = *pod.Spec.TerminationGracePeriodSeconds
+	}
 	switch pod.Status.Phase {
 	case v1.PodRunning:
-		if pod.DeletionTimestamp != nil {
+		if pod.DeletionTimestamp != nil &&
+			time.Now().Unix()-pod.DeletionTimestamp.Unix() <= gracePeriodSeconds {
 			return Releasing
 		}
 
 		return Running
 	case v1.PodPending:
-		if pod.DeletionTimestamp != nil {
+		if pod.DeletionTimestamp != nil &&
+			time.Now().Unix()-pod.DeletionTimestamp.Unix() <= gracePeriodSeconds {
```

Comment:
`time.Now().Unix()-pod.DeletionTimestamp.Unix()`will keep growing, If once the time point of less than or equal to is missed, then this judgment will always be false and will never return release. Is this what we want?

Reply:
it means this pod is always exists with terminating status if `time.Now().Unix()-pod.DeletionTimestamp.Unix()`keep growing, resource cannot be freed. this resource is always allocated for scheduler.

Reply:
> `time.Now().Unix()-pod.DeletionTimestamp.Unix()`will keep growing, If once the time point of less than or equal to is missed, then this judgment will always be false and will never return release. Is this what we want?

I think gracePeriodSeconds can set bigger,  it should be another status not "Running" status when waiting time bigger than gracePeriodSeconds, but I have no idea what status to set

Reply:
Sorry I don't think this is a perfect fix

Reply:
I think should add a status  like "ReleaseFailed", in the code of resource calculation, it should be the same as Running. what do you think?

Reply:
@wangyang0616 
Can you help review this PR

Reply:
yeah, I'll close it and fix it with add staus. when I finish it I'll reopen it again